### ISSUE TITLE
Needs a new Hackage release to suppiort ghc 9.10

### ISSUE THREAD

**Issue by erikd** [bgamari/monoidal-containers#106]
Building `monoidal-containers-0.6.4.0` with `ghc-9.10.1` I get:
```
src/Data/IntMap/Monoidal.hs:54:7: error: [GHC-87543]
    Ambiguous occurrence ‘foldl'’.
    It could refer to
       either ‘Prelude.foldl'’,
              imported from ‘Prelude’ at src/Data/IntMap/Monoidal.hs:130:1-63
              (and originally defined in ‘ghc-internal-9.1001.0:GHC.Internal.Data.Foldable’),
           or ‘Data.IntMap.Monoidal.foldl'’,
              defined at src/Data/IntMap/Monoidal.hs:473:1.
   |
54 |     , foldl'
   |       ^^^^^^
```
where as building git `HEAD` (`c838b6cebe40bb07512ed19bca9881901938396b`) with `ghc-9.10.1` has zero compile errors.

We need what is currently at `HEAD` to be released as a new version.

**Comment by alexfmpe** [comment]
Likely fixed by #105

**Comment by erikd** [comment]
@alexfmpe @ali-abrar Sorry, it is not fixed by #105. 

The only fix for this is to upload a new package based on git `HEAD`.

**Comment by ali-abrar** [comment]
Uploaded.
